schemes/chemistry/prescribed_aerosols.F90

```diff
@@ -0,0 +1,654 @@
+    prescribed_aero_model, &
+    aerosol_constituents, &
+    errmsg, errflg)
+
+    use ccpp_constituent_prop_mod, only: ccpp_constituent_properties_t
+
+    ! Input arguments:
+    logical,            intent(in)  :: amIRoot
+    integer,            intent(in)  :: iulog
+    character(len=*),   intent(in)  :: prescribed_aero_specifier(:)     ! aerosol specifiers from namelist
+    character(len=*),   intent(in)  :: prescribed_aero_file             ! input filename from namelist
+    character(len=*),   intent(in)  :: prescribed_aero_filelist         ! input filelist from namelist
+    character(len=*),   intent(in)  :: prescribed_aero_datapath         ! input datapath from namelist
+    character(len=*),   intent(in)  :: prescribed_aero_type             ! data type from namelist
+    integer,            intent(in)  :: prescribed_aero_cycle_yr         ! cycle year from namelist [1]
+    integer,            intent(in)  :: prescribed_aero_fixed_ymd        ! fixed year-month-day from namelist (YYYYMMDD) [1]
+    integer,            intent(in)  :: prescribed_aero_fixed_tod        ! fixed time of day from namelist [s]
+    character(len=*),   intent(in)  :: prescribed_aero_model            ! type of aerosol representation [none]
+
+    ! Output arguments:
+    type(ccpp_constituent_properties_t), allocatable, intent(out) :: aerosol_constituents(:) ! prescribed aero runtime CCPP constituents
+    character(len=512), intent(out) :: errmsg
+    integer,            intent(out) :: errflg
+
+    ! Local variables:
+    character(len=256) :: tmpstr
+    character(len=64)  :: constituent_name
+    character(len=64)  :: ncdf_fld_name
+    character(len=64)  :: unit_name
+    integer            :: strlen
+    integer            :: idx, idx2
+    integer            :: aero_idx
+    integer            :: i, j
+    logical            :: is_modal_aero_interstitial
+    logical            :: skip_spec
+
+    character(len=*), parameter :: subname = 'prescribed_aerosols_register'
+
+    errmsg = ''
+    errflg = 0
+
+    ! Store module-level settings
+    clim_modal_aero = (trim(prescribed_aero_model) == "modal")
+
+    ! Check if prescribed aerosols are enabled
+    if (prescribed_aero_file == 'UNSET' .or. &
+        len_trim(prescribed_aero_file) == 0) then
+      has_prescribed_aerosols = .false.
+      if (amIRoot) then
+        write(iulog,*) subname//': No prescribed aerosols specified'
+      end if
+      return
+    end if
+
+    ! Parse the aerosol format specifier from namelist into mapping ddt.
+    ! We need two scans. First to determine the count, the second to populate
+    ! the information into the ddt.
+    aero_cnt   = 0
+    aero_cnt_c = 0 ! cloud borne species count
+    cnt_loop: do i = 1, N_AERO_MAX
+      ! FIXME: should I be responsible for handling this? I feel like I should not handle this
+      if(prescribed_aero_specifier(i) == 'UNSET' .or. &
+         len_trim(prescribed_aero_specifier(i)) == 0) exit cnt_loop
```

Comment:
I wonder if we could simplify this, and get rid of the hardcoded value for `N_AERO_MAX`, by just determining the total number of non-empty and non-UNSET elements in `prescribed_aero_specifier`?  I think that can be done like so:


```suggestion
    n_aero_max = count((prescribed_aero_specifier /= 'UNSET') .and. (len_trim(prescribed_aero_specifier) > 0)) 
    cnt_loop: do i = 1, n_aero_max      
```